homeassistant/components/media_player/emby.py

```diff
@@ -0,0 +1,377 @@
+"""
+Support to interface with the Emby API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.emby/
+"""
+import logging
+
+from datetime import timedelta
+
+import uuid
+import requests
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.media_player import (
+    MEDIA_TYPE_TVSHOW, MEDIA_TYPE_VIDEO, SUPPORT_NEXT_TRACK, SUPPORT_PAUSE,
+    SUPPORT_SEEK, SUPPORT_STOP, SUPPORT_PREVIOUS_TRACK, MediaPlayerDevice,
+    PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_HOST, CONF_API_KEY, CONF_PORT, DEVICE_DEFAULT_NAME, STATE_IDLE,
+    STATE_OFF, STATE_PAUSED, STATE_PLAYING, STATE_UNKNOWN, CONTENT_TYPE_JSON,
+    MAJOR_VERSION, MINOR_VERSION)
+from homeassistant.helpers.event import (track_utc_time_change)
+from homeassistant.util import Throttle
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
+
+DEFAULT_PORT = 8096
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_EMBY = SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | \
+    SUPPORT_STOP | SUPPORT_SEEK
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+})
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Setup the Emby platform."""
+    host = config.get(CONF_HOST)
+    key = config.get(CONF_API_KEY)
+    port = config.get(CONF_PORT)
+
+    url = 'http://{}:{}'.format(host, port)
```

Comment:
Good point.  I'll move the protocol definition to be defined in the configuration so either http or https can be specified.